snapcraft.yaml

```diff
@@ -34,6 +34,7 @@ apps:
     daemon: simple
     plugs:
       - block-devices
+      - dm-crypt
```

Comment:
Is that a new interface to snapd?

If so, we'll probably want to introduce a `assumes` section so we get snapd to fail ahead of time with an error that's more user friendly.

Reply:
Frustratingly, it exists (not sure since when) but isn't complete pending a snapd release

Reply:
Okay, so will probably want to still do an `assumes` for the release where it actually works.
Will also need to file an auto-connect request for that interface as I don't expect the store to grant it automatically.

Reply:
the code for the encryption should actually handle the case where the interface is present but not functional, it just doesn't allow encryption then

Reply:
Ah, okay, that's good.

Reply:
specifically, this bit: https://github.com/canonical/microceph/pull/77/files#diff-3418d4d1a1b4a82b23ce7f4b04ec10f2fff558f4e61ef25e2f4f93834d33738cR192